src/utils/schema-validator.js

```diff
@@ -1,15 +1,39 @@
+
 const fs = require('fs');
 const path = require('path');
+const SwaggerClient = require('swagger-client');
 const Validator = require('swagger-model-validator');
 const yaml = require('js-yaml');
 
-const validateRequest = (request, schemaName) => {
-  const specPath = path.resolve(__dirname, '..', 'specs', 'api.yaml');
-  const specObj = yaml.safeLoad(fs.readFileSync(specPath), 'utf8');
-  const validator = new Validator();
+const validateRequest = async (request, schemaPath) => {
+  const specPath = path.resolve(__dirname, '..', 'specs', 'models', schemaPath);
+
+  // Create a custom Swagger client and 'HTTP fetcher' mock so we can load in
+  // our spec spread across multiple local files. The result is the entire spec
+  // fully resolved across all refs and imports.
+  const spec = await SwaggerClient.resolve({
+    url: specPath,
+    http: async ({ url, headers }) => {
+      const data = yaml.safeLoad(fs.readFileSync(specPath), 'utf8');
 
+      return {
+        ok: true,
+        status: 200,
+        statusText: 'OK',
+        url,
+        headers,
+        text: JSON.stringify(data),
+        data: JSON.stringify(data),
+        body: data,
+        obj: data,
+      };
+    },
+  });
```

Comment:
Since `swagger-js` does not support local files, I had to write a mock HTTP client that will open local files, read their contents, and return an OK response. This ensures we can split our files and still validate work requests as necessary.

Reply:
By "local files" do you mean local $refs?

Reply:
Is there a reason to having the keys `text` and `data` with the same value?